R/bootstrap.R

```diff
@@ -43,19 +43,27 @@ bootstrapPage <- function(..., title = NULL, responsive = deprecated(), theme =
   }
 
   ui <- tagList(
-    bootstrapLib(theme),
```

Comment:
This can be replaced with:

```R
if (is_bs_theme(theme)) bootstrapLib(theme),
```

(See comment below for why we'd do it this way)

Reply:
I think we'd still have the same problem because of this?

```r
> str(tagList(NULL, "a"))
List of 2
 $ : NULL
 $ : chr "a"
 - attr(*, "class")= chr [1:2] "shiny.tag.list" "list"
```

Reply:
I suppose we could do a `do.call(tagList, dropNulls(ui))` though?

Reply:
Actually I'm not sure that's a good idea either since that could break existing code for similar reasons

Reply:
Oh right, the `NULL` would be in there. Something like this could work and is reasonably straightforward:

```R
  args <- list(
    if (!is.null(title)) tags$head(tags$title(title)),
    if (is.character(theme)) {
      if (length(theme) > 1) stop("`theme` must point to a single CSS file, not multiple files.")
      tags$head(tags$link(rel="stylesheet", type="text/css", href=theme))
    },
    # remainder of tags passed to the function
    list(...)
  )

  if (is_bs_theme(theme)) {
    args <- c(bootstrapLib(theme), args)
    ui <- do.call(tagList, args)
  } else {
    ui <- do.call(tagList, args)
    ui <- attachDependencies(ui, bootstrapLib())
  }
```

Reply:
I landed on something slightly different before seeing this e117d7e

Would you prefer to do the `do.call()` in the non-bslib case as well